lugenx/ecohabit
# Implement authentication functionality

this pull request implements the authentication functionality for the application by communicating with the back-end with already created endpoints. the following changes have been made:

- corrected the naming of some variables for consistency
- implemented registration, login, and logout functionality
- added and used Context API for state management.
- added registration success message
- added login failure messages

how to test:

to test the changes, follow these steps:

1. click on the "Sign up" button and fill out the registration form.
2. once you've registered, you should get a success message and be directed to the login page. 
3. try logging in using the credentials you just created.
4. if the login is successful, you should be redirected to the main page. If the login fails, you should see error messages.
5. to log out, click on the "Logout" button.
